### PR TITLE
Workflow: add details for review

### PR DESCRIPTION
**Pull request by 0xc0170**
Two main changes:

- every change in pull request should result in a comment (PR rebased, added new test, more devices support added, rewrote the history, etc) - helps keep tracking and the intention of the update
- another one is about rereview - if a change was approved by a technical person, and no change from technical point of view, then maintainers approval should be sufficient. For instance, documentation update, rebase on top of master, rewriting history, style changes and similar.

@ARMmbed/mbed-os-maintainers @bulislaw Please review

### CODE REVIEW

**Comment by iriark01** [comment]
@jen3andruska

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -160,6 +161,8 @@ Each state is time boxed. In most cases, this is sufficient time to move to anot
 
 All pull requests must be reviewed. The Arm Mbed CI bot determines the most suitable person to review the pull request and tags that person accordingly.
 
+Github dismissed reviewers status after any change to the pull request commit history (adding new commit, rebase). Simple changes like documentation edits or rebase on top of latest master should only require rereview by maintainers. Their approval is sufficient considering it was already approved by a team assigned as a reviewer.
```
Comment (jen3andruska) [inline docs/reference/contributing/guidelines/workflow.md]: Should this be 'dismisses' with an 's'?  Confused about the change to past tense here.

**Comment by AnotherButler** [comment]
@jen3andruska Thanks for the awesome edits 👍

**Comment by jen3andruska** [comment]
@AnotherButler No problem, happy to help 👍

**Pull request merged**